apache/superset
# fix(celery cache warmup): add auth and use warm_up_cache endpoint

### SUMMARY

Fixes cache warmup via celery job. 

- adds cache and data cache configs to the docker-compose superset_config to enable caching for development env.
- add authentication to the cache warmup celery beat job, using the `THUMBNAIL_SELENIUM_USER` config for auth. 
- use the `/superset/warm_up_cache` endpoint for warming up the cache. 
- run the warmup call as a single celery job instead of running all the warmups in single job. This approach should enable some parallelization of the warm up calls. A slow warmup call should also not slow down the entire process. 

There is currently an [open PR](https://github.com/apache/superset/pull/20387) that addresses this issue, however it is using a web driver process to compute the warmup by visiting the dashboard which seems a bit process heavy. Since there is already a cache warmup endpoint that is able to handle charts and filters combos, it make more sense to use that endpoint. 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF


### TESTING INSTRUCTIONS


- use the docker-compose workflow
- Add a celery cache warmup beat config using `top_n_strategy`, set it to run at a high interval (for eg, every minute). documented [here](https://github.com/apache/superset/blob/5113b01031705128df2064068a0809f07019c8ae/superset/tasks/cache.py#L96)
- visit dashboards to that top n strategy has data to work with
- observe the cache warmup logs in the worker 
- visit dashboards and notice the charts are cached

### ADDITIONAL INFORMATION


- [ ] Has associated issue: Fixes #18933, Fixes #9597
